python/lsst/daf/butler/registries/sqliteRegistry.py

```diff
@@ -47,5 +47,27 @@ class SqliteRegistry(SqlRegistry):
         Load configuration
     """
 
+    @classmethod
+    def setConfigRoot(cls, root, config, full):
+        """Set any filesystem-dependent config options for this Registry to
+        be appropriate for a new empty repository with the given root.
+
+        Parameters
+        ----------
+        root : `str`
+            Filesystem path to the root of the data repository.
+        config : `Config`
+            A Butler-level config object to update (but not a
+            `ButlerConfig`, to avoid included expanded defaults).
+        full : `ButlerConfig`
+            A complete Butler config with all defaults expanded;
+            repository-specific options that should not be obtained
+            from defaults when Butler instances are constructed
+            should be copied from `full` to `Config`.
+        """
+        config["registry.db"] = "sqlite:///{}/gen3.sqlite3".format(root)
+        for key in ("registry.cls",):
```

Comment:
I do wonder whether we could save some code duplication by having these keys as class attributes and then a routine that reads the keys and copies them (and maybe attempts to run `format(root)` on them.

Reply:
Perhaps a bit, but I'm content to wait until we see what other concrete Datastore/Registry implementations actually need to do here before worrying about them duplicating code.